app/controllers/transactions_controller.rb

```diff
@@ -1,63 +1,75 @@
+# frozen_string_literal: true
+
 class TransactionsController < ApplicationController
-  before_action :set_transaction, only: [:edit, :update, :destroy, :show]
+  before_action :authenticate_user!
+  before_action :set_transaction, only: %i[edit update destroy show]
 
   def index
```

Comment:
Create a service object for listing transactions. Something like below:

```
class ListTransactions
  def initialize(page:, per_page, user:)
    @page = page
    @per_page = per_page
    @user = user
  end

  def call
  end
end

```

and call here so that index method looks like this

```
def index
  @transactions = ListTransactions.new(page: params[:page], per_page: 5, user: current_user).call
end
```